src/tools/ceph_objectstore_tool.cc

```diff
@@ -2563,6 +2563,44 @@ int set_size(
   return 0;
 }
 
+int clear_data_digest(ObjectStore *store, coll_t coll, ghobject_t &ghobj, ObjectStore::Sequencer &osr) {
+  if (ghobj.hobj.is_snapdir()) {
+    cerr << "Can't clear data digest of a snapdir" << std::endl;
+    return -EINVAL;
+  }
+  bufferlist attr;
+  int r = store->getattr(coll, ghobj, OI_ATTR, attr);
+  if (r < 0) {
+    cerr << "Error getting attr on : " << make_pair(coll, ghobj) << ", "
+       << cpp_strerror(r) << std::endl;
+    return r;
+  }
+  object_info_t oi;
+  bufferlist::iterator bp = attr.begin();
+  try {
+    ::decode(oi, bp);
+  } catch (...) {
+    r = -EINVAL;
+    cerr << "Error getting attr on : " << make_pair(coll, ghobj) << ", "
+         << cpp_strerror(r) << std::endl;
+    return r;
+  }
+  if (!dry_run) {
+    attr.clear();
+    oi.clear_data_digest();
+    ::encode(oi, attr, -1);
```

Comment:
Other users of encode() with -1 as the features add this comment "/* fixme: using full features */"

Reply:
In set_size() above the encode looks like this:

`encode(oi, attr, -1);  /* fixme: using full features */`